File: algorithms/9. Palindrome Number/solution.cc
class Solution {
public:
    bool isPalindrome(int x) {
        long sum = 0;
        int n = x;
        while ( n > 0 ) {
            sum = sum * 10 + n % 10;
            n /= 10;
        }
        return x == sum;
    }
};

File: algorithms/9. Palindrome Number/solution.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        r, n = 0, x
        while n > 0:
            r = r * 10 + n % 10
            n /= 10
        return x == r
